src/modules/user/user.controller.spec.ts

```diff
@@ -0,0 +1,157 @@
+
+  beforeEach(async () => {
+    userUseCases = createMock<UserUseCases>();
+    notificationService = createMock<NotificationService>();
+    keyServerUseCases = createMock<KeyServerUseCases>();
+    cryptoService = createMock<CryptoService>();
+    sharingService = createMock<SharingService>();
+
+    userController = new UserController(
+      userUseCases,
+      notificationService,
+      keyServerUseCases,
+      cryptoService,
+      sharingService,
+    );
+  });
+
+  it('should be defined', () => {
+    expect(userController).toBeDefined();
+  });
+
+  describe('POST /unblock-account', () => {
+    it('When user is not found, then returns NotFoundException', async () => {
+      userUseCases.sendAccountUnblockEmail.mockRejectedValueOnce(
+        new NotFoundException(),
+      );
+      await expect(
+        userController.requestAccountUnblock({ email: 'test@test.com' }),
+      ).rejects.toThrow(NotFoundException);
+    });
+    it('When an unexpected error is throw, then returns InternalServerException', async () => {
+      userUseCases.sendAccountUnblockEmail.mockRejectedValueOnce(new Error());
+      await expect(
+        userController.requestAccountUnblock({ email: 'test@test.com' }),
+      ).rejects.toThrow(InternalServerErrorException);
+    });
+  });
+
+  describe('PUT /unblock-account', () => {
+    const user = newUser();
+    const validToken = SignWithCustomDuration(
+      {
+        payload: {
+          uuid: user.uuid,
+          email: user.email,
+          action: AccountTokenAction.Unblock,
+        },
+      },
+      getEnv().secrets.jwt,
+      '48h',
+    );
+    it('When token has invalid signature, then fails', async () => {
+      const invalidToken = SignWithCustomDuration(
+        {
+          payload: {},
+        },
+        'Invalid Signature',
+        '48h',
+      );
+      await expect(userController.accountUnblock(invalidToken)).rejects.toThrow(
+        ForbiddenException,
+      );
+    });
+
+    it('When token has valid signature but incorrect properties, then fails', async () => {
+      const invalidToken = SignWithCustomDuration(
+        {
+          payload: {
+            uuid: 'invalid Uuid',
+            email: 'test@test.com',
+            action: 'not unlock action',
+          },
+        },
+        getEnv().secrets.jwt,
+        '48h',
+      );
+      await expect(userController.accountUnblock(invalidToken)).rejects.toThrow(
+        ForbiddenException,
+      );
+    });
+
+    it('When token is expired, then fails', async () => {
+      const expiredToken = SignWithCustomDuration(
+        {
+          payload: {},
+        },
+        getEnv().secrets.jwt,
+        '-48h',
+      );
+      await expect(userController.accountUnblock(expiredToken)).rejects.toThrow(
+        ForbiddenException,
+      );
+    });
+
+    it('When token is valid but useCase throws badRequest or Forbidden, then fails with respective error', async () => {
```

Comment:
Why is this important? I mean, wouldn't this be a unique test that tells us that the errors are propagated instead of one test per each type of error?